pkg/reconciler/upgrade.go

```diff
@@ -0,0 +1,39 @@
+/*
+Copyright 2020 The Knative Authors
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    https://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package reconciler
+
+import (
+	"fmt"
+
+	apierrs "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/kubernetes"
+	"knative.dev/pkg/system"
+)
+
+// RemovePreUnified resources removes resources that predate the fusion of the operators
+// into a single binary.
+func RemovePreUnifiedResources(kube kubernetes.Interface, oldName string) error {
+	if err := kube.AppsV1().Deployments(system.Namespace()).Delete(oldName, &metav1.DeleteOptions{}); err != nil && !apierrs.IsNotFound(err) {
+		return fmt.Errorf("failed to remove old deployment %s: %w", oldName, err)
+	}
+	if err := kube.CoreV1().ServiceAccounts(system.Namespace()).Delete(oldName, &metav1.DeleteOptions{}); err != nil && !apierrs.IsNotFound(err) {
+		return fmt.Errorf("failed to remove old serviceaccount %s: %w", oldName, err)
+	}
+
```

Comment:
Other resources, like roles(knative-serving-operator-aggregated, knative-serving-operator, knative-serving-eventing-aggregated and knative-eventing-operator), and rolebindings(knative-serving-operator-aggregated, knative-serving-operator, knative-serving-eventing-aggregated and knative-eventing-operator), ca also be removed.
I think I have listed them all.

Reply:
They can not, no. I left them all in. They now just aggregate to a different ServiceAccount.